certora/specs/ChangesMorphoMarketV1Adapter.spec

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (c) 2025 Morpho Association
+
+using Utils as Utils;
+
+methods {
+    function allocation(Morpho.MarketParams) external returns (uint256) envfree;
+
+    // Needed because linking fails.
+    function _.transfer(address, uint256) external => DISPATCHER(true);
+    function _.transferFrom(address, address, uint256) external => DISPATCHER(true);
+
+    function _.borrowRate(Morpho.MarketParams, Morpho.Market) external => constantBorrowRate expect uint256;
+    function _.borrowRateView(Morpho.MarketParams, Morpho.Market) external => constantBorrowRate expect uint256;
+
+    function Utils.decodeMarketParams(bytes) external returns (Morpho.MarketParams) envfree;
+}
+
+persistent ghost uint256 constantBorrowRate;
+
+// Check that calling allocate or deallocate with 0 amount yields the same change.
+rule sameChangeForAllocateAndDeallocateOnZeroAmount(env e, bytes data, bytes4 selector, address sender) {
+  storage initialState = lastStorage;
+
+  bytes32[] idsAllocate; int256 changeAllocate;
+  idsAllocate, changeAllocate = allocate(e, data, 0, selector, sender) at initialState;
```

Comment:
at this point what is the difference between having `at initialState` and not having it?

Reply:
No difference, I removed it if it makes things clearer